example/src/main/java/com/stripe/example/activity/AlipayPaymentNativeActivity.kt

```diff
@@ -0,0 +1,126 @@
+    private var confirmed = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(viewBinding.root)
+
+        viewBinding.confirmWithPaymentButton.text =
+            resources.getString(R.string.confirm_alipay_button)
+        viewBinding.paymentExampleIntro.text =
+            resources.getString(R.string.alipay_example_intro)
+
+        viewModel.inProgress.observe(this) { enableUi(!it) }
+        viewModel.status.observe(this, Observer(viewBinding.status::setText))
+
+        viewBinding.confirmWithPaymentButton.setOnClickListener {
+            clientSecret?.let {
+                // If we already loaded the Payment Intent and haven't confirmed, try again
+                if (!confirmed) {
+                    updateStatus("\n\nPayment Intent already created, trying to confirm")
+                    confirmPayment(it)
+                }
+            } ?: run {
+                createAndConfirmPaymentIntent(
+                    country = "US",
+                    paymentMethodCreateParams = PaymentMethodCreateParams.createAlipay(),
+                    supportedPaymentMethods = "alipay"
+                )
+            }
+        }
+    }
+
+    override fun handleCreatePaymentIntentResponse(
+        responseData: JSONObject,
+        params: PaymentMethodCreateParams?,
+        shippingDetails: ConfirmPaymentIntentParams.Shipping?,
+        stripeAccountId: String?,
+        existingPaymentMethodId: String?,
+        mandateDataParams: MandateDataParams?,
+        onPaymentIntentCreated: (String) -> Unit
+    ) {
+        viewModel.status.value +=
+            "\n\nStarting PaymentIntent confirmation" +
+            (
+                stripeAccountId?.let {
+                    " for $it"
+                } ?: ""
+                )
+
+        clientSecret = responseData.getString("secret").also {
+            confirmPayment(it)
+        }
+    }
+
+    private fun confirmPayment(clientSecret: String) {
+        stripe.confirmAlipayPayment(
+            confirmPaymentIntentParams = ConfirmPaymentIntentParams.createAlipay(clientSecret),
+            authenticator = { data ->
+                PayTask(this).payV2(data, true)
+            },
+            callback = object : ApiResultCallback<PaymentIntentResult> {
+                override fun onSuccess(result: PaymentIntentResult) {
+                    val paymentIntent = result.intent
+                    when (paymentIntent.status) {
+                        StripeIntent.Status.Succeeded -> {
+                            confirmed = true
+                            updateStatus("\n\nPayment succeeded")
+                        }
+                        StripeIntent.Status.RequiresAction ->
+                            updateStatus("\n\nUser canceled confirmation")
+                        else ->
+                            updateStatus(
+                                "\n\nPayment failed or canceled." +
```

Comment:
It seems last time I checked, an in-app web browser would be opened if we cancel the native flow, seems that's no longer happening with the recording, could you elaborate a bit why that happened?

Reply:
I removed [line 32 here](https://stripe.com/docs/payments/alipay/accept-a-payment?platform=mobile&ui=android#android-redirect-alipay-wallet), and have a PR updating the documentation.
When the user completes the payment in the Alipay app, the PI status will be `Suceeded`. If they cancel, it will be `RequiresAction`.
Calling `Stripe.handleNextActionForPayment` would try to confirm again, using a webview. That's not what we want, since the user just canceled. The docs are also not showing how to handle the result of that call, which is handled in `onActivityResult`.